tests/benchmarks/sys_kernel/src/mem_slab.c

```diff
@@ -0,0 +1,107 @@
+/* memslab.c */
+
+/*
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "syskernel.h"
+
+#define MEM_SLAB_BLOCK_SIZE  (8)
+#define MEM_SLAB_BLOCK_CNT   (NUMBER_OF_LOOPS)
+#define MEM_SLAB_BLOCK_ALIGN (4)
+
+K_MEM_SLAB_DEFINE(my_slab,
+		  MEM_SLAB_BLOCK_SIZE,
+		  MEM_SLAB_BLOCK_CNT,
+		  MEM_SLAB_BLOCK_ALIGN);
+
+/* Array contains pointers to allocated regions. */
+void *slab_array[MEM_SLAB_BLOCK_CNT];
+
+/**
+ *
+ * @brief Memslab allocation test function.
+ *		  This test allocates available memory space.
+ *
+ * @param no_of_loops  Amount of loops to run.
+ * @param test_repeats Amount of test repeats per loop.
+ *
+ * @return NUmber of done loops.
+ */
+static int mem_slab_alloc_test(int no_of_loops)
+{
+	int i;
+
+	for (i = 0; i < no_of_loops; i++) {
+		if (k_mem_slab_alloc(&my_slab, &slab_array[i], K_MSEC(100))
```

Comment:
Wouldn't `K_NO_WAIT` make more sense in the benchmarking context?

Reply:
Changed.